### ISSUE TITLE
[FEATURE REQUEST] Inline fullscreen

### ISSUE THREAD

**Issue by tagerjump** [sarbagyastha/youtube_player_flutter#168]
While in portrait mode this plugin provides a **inline** player, which is very nice. However, the approach for toggling fullscreen now is to push a full screen page. I'm not able to **build my own widgets in fullscreen mode**. The Youtube official app implement full screen mode by just setting PreferredOrientations to [DeviceOrientation.landscapeLeft,]. 

#### This is what I want to achieve
![RPReplay_Final1583528918](https://user-images.githubusercontent.com/45743053/76124565-f82cb980-5fc8-11ea-806f-14990e41d7ba.gif)

I tried to do it the same way list below, but the player will be paused due to resize. #50 
It paused and stucked for a few seconds before continue playing when set to landscape(fullscreen) mode, which is not good but still acceptable to me. But errors will occur when go back to portrait mode and won't continue playing.
Please notice that I'm using my own playerUI without calling the toggleFullScreen() method.
```dart
    _subscription = OrientationPlugin.onOrientationChange.listen((value) async{
      if (!mounted) return;
      if(value == DeviceOrientation.portraitUp){
        setState(() {
          _deviceOrientation = 0;
        });
        // Notice player to hide/unhide video description before setPreferredOrientations()
        widget.notifyParentOrientation(_deviceOrientation);
        await SystemChrome.setPreferredOrientations([DeviceOrientation.portraitUp,]);

      }
      else if(value == DeviceOrientation.landscapeRight){
        setState(() {
          _deviceOrientation = 1;
        });
        widget.notifyParentOrientation(_deviceOrientation);
        await SystemChrome.setPreferredOrientations([DeviceOrientation.landscapeLeft,]);

      }
      else if(value == DeviceOrientation.landscapeLeft){
        setState(() {
          _deviceOrientation = 3;
        });
        widget.notifyParentOrientation(_deviceOrientation);
        await SystemChrome.setPreferredOrientations([DeviceOrientation.landscapeRight,]);
      }
    });
```
```dart
    return Scaffold(
      body: Column(
        children: <Widget>[
          youtubePlayer,    // the YoutubePlayer widget
          // The widgets below will be offstage in fullscreen mode
          Offstage(
            offstage: (_deviceOrientation == 1 || _deviceOrientation == 3),
            child: ConstrainedBox(
              constraints: BoxConstraints(
                maxHeight: _deviceOrientation == 0 ? (1 - heightFraction) * height : 0,
              ),
              child: VideoTabView(widget.id, widget.root, videoTabKey),
            ),
          )
          //(_deviceOrientation == 1 || _deviceOrientation == 3) ?
          //  Container()
          //:
        ],
      ),
    );
```

Exceptions occur when go back to portrait mode.
```
E/InputMethodManager( 5517): b/117267690: Failed to get fallback IMM with expected displayId=7 actual IMM#displayId=0 view=io.flutter.plugins.webviewflutter.InputAwareWebView{962e9fd VFEDHVC.. ........ 0,0-2028,1141}
D/eglCodecCommon( 5517): setVertexArrayObject: set vao to 0 (0) 0 2
D/EGL_emulation( 5517): eglMakeCurrent: 0xdc01a780: ver 3 0 (tinfo 0xdc00f0b0)

════════ (3) Exception caught by foundation library ════════════════════════════════════════════════
The method 'play' was called on null.
Receiver: null
Tried calling: play()
════════════════════════════════════════════════════════════════════════════════════════════════════
D/EGL_emulation( 5517): eglMakeCurrent: 0xdc01b020: ver 3 0 (tinfo 0xdc00ff90)
D/EGL_emulation( 5517): eglMakeCurrent: 0xdc01a780: ver 3 0 (tinfo 0xdc00f0b0)
D/eglCodecCommon( 5517): setVertexArrayObject: set vao to 0 (0) 23 2
D/        ( 5517): PlayerBase::stop() from IPlayer
D/AudioTrack( 5517): stop(12): called with 464359 frames delivered
D/EGL_emulation( 5517): eglMakeCurrent: 0xdc01a780: ver 3 0 (tinfo 0xdc00f0b0)

════════ (4) Exception caught by foundation library ════════════════════════════════════════════════
The method 'play' was called on null.
Receiver: null
Tried calling: play()
════════════════════════════════════════════════════════════════════════════════════════════════════

════════ (5) Exception caught by foundation library ════════════════════════════════════════════════
The method 'play' was called on null.
Receiver: null
Tried calling: play()
════════════════════════════════════════════════════════════════════════════════════════════════════

════════ (6) Exception caught by foundation library ════════════════════════════════════════════════
The method 'play' was called on null.
Receiver: null
Tried calling: play()
════════════════════════════════════════════════════════════════════════════════════════════════════

════════ (7) Exception caught by foundation library ════════════════════════════════════════════════
The method 'play' was called on null.
Receiver: null
Tried calling: play()
════════════════════════════════════════════════════════════════════════════════════════════════════
D/EGL_emulation( 5517): eglMakeCurrent: 0xdc01a780: ver 3 0 (tinfo 0xdc00f0b0)

════════ (8) Exception caught by foundation library ════════════════════════════════════════════════
The method 'play' was called on null.
Receiver: null
Tried calling: play()
════════════════════════════════════════════════════════════════════════════════════════════════════

════════ (9) Exception caught by foundation library ════════════════════════════════════════════════
The method 'play' was called on null.
Receiver: null
Tried calling: play()
════════════════════════════════════════════════════════════════════════════════════════════════════

════════ (10) Exception caught by foundation library ═══════════════════════════════════════════════
The method 'play' was called on null.
Receiver: null
Tried calling: play()
════════════════════════════════════════════════════════════════════════════════════════════════════

════════ (11) Exception caught by foundation library ═══════════════════════════════════════════════
The method 'play' was called on null.
Receiver: null
Tried calling: play()
════════════════════════════════════════════════════════════════════════════════════════════════════

════════ (12) Exception caught by foundation library ═══════════════════════════════════════════════
The method 'play' was called on null.
Receiver: null
Tried calling: play()
════════════════════════════════════════════════════════════════════════════════════════════════════
D/eglCodecCommon( 5517): setVertexArrayObject: set vao to 0 (0) 0 0
D/EGL_emulation( 5517): eglMakeCurrent: 0xc4bb36e0: ver 3 0 (tinfo 0xc4c96ff0)
E/eglCodecCommon( 5517): glUtilsParamSize: unknow param 0x000088ef
E/eglCodecCommon( 5517): glUtilsParamSize: unknow param 0x000088ef
D/eglCodecCommon( 5517): setVertexArrayObject: set vao to 22 (22) 0 0
D/EGL_emulation( 5517): eglMakeCurrent: 0xdc01a780: ver 3 0 (tinfo 0xdc00f0b0)
E/eglCodecCommon( 5517): glUtilsParamSize: unknow param 0x000088ef
E/eglCodecCommon( 5517): glUtilsParamSize: unknow param 0x000088ef
D/eglCodecCommon( 5517): setVertexArrayObject: set vao to 8 (8) 23 0
D/EGL_emulation( 5517): eglMakeCurrent: 0xdc01a780: ver 3 0 (tinfo 0xdc00f0b0)
D/eglCodecCommon( 5517): setVertexArrayObject: set vao to 0 (0) 23 2
I/chromium( 5517): [INFO:CONSOLE(83)] "Uncaught TypeError: Cannot read property 'playVideo' of undefined", source: https://www.youtube.com/ (83)
D/EGL_emulation( 5517): eglMakeCurrent: 0xdc01a780: ver 3 0 (tinfo 0xdc00f0b0)
D/eglCodecCommon( 5517): setVertexArrayObject: set vao to 8 (8) 23 58
D/EGL_emulation( 5517): eglMakeCurrent: 0xdc01a780: ver 3 0 (tinfo 0xdc00f0b0)
D/eglCodecCommon( 5517): setVertexArrayObject: set vao to 0 (0) 23 2
I/chromium( 5517): [INFO:CONSOLE(139)] "Unrecognized feature: 'picture-in-picture'.", source: https://s.ytimg.com/yts/jsbin/www-widgetapi-vflkAsU_-/www-widgetapi.js (139)
D/eglCodecCommon( 5517): setVertexArrayObject: set vao to 17 (17) 0 0
D/EGL_emulation( 5517): eglMakeCurrent: 0xdc01a780: ver 3 0 (tinfo 0xdc00f0b0)
D/eglCodecCommon( 5517): setVertexArrayObject: set vao to 8 (8) 23 56
```

I think setPreferredOrientations is the best way if developer want to keep a inline Youtube-style player while in fullscreen mode. Is there any way to prevent pause and errors when set orientation?

**Comment by tagerjump** [comment]
I found that the errors were not cause by the YoutubePlayer.
I modify the code like and it seems to be okay
```dart
    double height = MediaQuery.of(context).size.height;
    return Scaffold(
        body: Column(
          children: <Widget>[
            LimitedBox(
              maxHeight: height,
              child: playerStack,
            ),
            Expanded(
              child: Offstage(
                  offstage: _deviceOrientation != 0,
                  child: videoTabView
              ),
            ),
          ],
        )
    );
```
But the video pause or stop occasionally (not every time). I think this issue should be merged with #50.

**Comment by sarbagyastha** [comment]
Should fix with v7.0.0. Feel free to reopen if the issue persists.
